real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/simulation/LineByLineSensorPlayer.java

```diff
@@ -0,0 +1,112 @@
+package com.roboclub.robobuggy.simulation;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonObject;
+import com.roboclub.robobuggy.main.RobobuggyConfigFile;
+import com.roboclub.robobuggy.main.RobobuggyLogicNotification;
+import com.roboclub.robobuggy.main.RobobuggyMessageLevel;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Date;
+
+/**
+ *
+ * @author Trevor Decker
+ * a tool for reading log files line by line
+ *
+ */
+public class LineByLineSensorPlayer {
+
+	/**
+	 * Constructor for sensor player
+	 * @param filePath the file to play
+	 * @param playBackSpeed the playback sped
+	 */
+	public LineByLineSensorPlayer(String filePath,double playBackSpeed) {
+		//open up the log file 
+
+		 new RobobuggyLogicNotification("initializing the SensorPlayer", RobobuggyMessageLevel.NOTE);
+		 Thread thread = new Thread(){
+			    public void run(){
+			    	Gson translator = new GsonBuilder().create();//TODO don't create a new one for every call of this function
+			    	// open input stream test.txt for reading purpose.
+			    	try {
+			    		InputStream is = new FileInputStream(filePath);
+			    		// create new input stream reader
+			    		InputStreamReader isr = new InputStreamReader(is, "UTF-8");
+			    		//the file was created so lets try and read it
+			    		BufferedReader br = new BufferedReader(isr);
+			    		String nextLine;
+			    			nextLine= br.readLine();  //reads {
+						if(nextLine != null)
+			    			nextLine = br.readLine();  // name
+						else {
+							new RobobuggyLogicNotification("File wasn't a log file!", RobobuggyMessageLevel.EXCEPTION);
+							br.close();
+							return;
+						}
+						if(nextLine != null)
```

Comment:
If we don't plan on doing anything with the information at the top of the log file, can we rewrite this to be a loop?  That would make it easier and cleaner should we decide to append more information as a header.